lib/ja_serializer/builder/link.ex

```diff
@@ -27,6 +27,19 @@ defmodule JaSerializer.Builder.Link do
   end
 
   defp path_for_context(context, path) do
+    parsed_uri = URI.parse path
+    if parsed_uri.port do
+      replaced_path = replaced_path_for_context(context, parsed_uri.path)
+      absolute_replaced_path = "#{parsed_uri.scheme}://#{parsed_uri.host}:#{parsed_uri.port}#{replaced_path}"
+      [ absolute_replaced_path, parsed_uri.query ]
+      |> Enum.reject(&(is_nil(&1)))
+      |> Enum.join("?")
+    else
+      replaced_path_for_context(context, path)
+    end
```

Comment:
@MishaConway I am wondering if we should just always parse the ui and just run the replaced path on the path, something like this (untested):

```elixir
uri = URI.parse(path)
uri
|> Map.put(:path, replaced_path_for_context(context, uri.path))
|> Map.put(:query, replaced_path_for_context(context, uri.query))
|> URI.to_string
```

I think that may just cover all situations (except maybe using an atom in the url/domain/subdomain).